array_api_tests/test_creation_functions.py

```diff
@@ -199,9 +241,34 @@ def test_ones(shape, dtype):
     assert a.shape == shape, "ones() produced an array with incorrect shape"
     assert all(equal(a, full((), ONE, **kwargs))), "ones() array did not equal 1"
 
-# TODO: implement ones_like (requires hypothesis arrays support)
-def test_ones_like():
-    pass
+
+@given(
+    a=xps.arrays(
+        dtype=shared_dtypes,
+        shape=xps.array_shapes(),
```

Comment:
I think we need to keep the shapes strategy as the one defined in hypothesis helpers. This makes it easy to globally limit the array shape, e.g., if the default strategy generates shapes that use too much memory or are too slow. It also is necessary to work around this NumPy bug https://github.com/data-apis/array-api-tests/blob/master/array_api_tests/hypothesis_helpers.py#L102-L105. Actually, it would be much more convenient if the hypothesis array_shapes strategy had a max_size argument, which controlled the maximum prod(shape).

Reply:
I resolved this, thanks.

Just noting I personally wouldn't want to implement a `max_size` arg in `xps.array_shapes()`, especially if one can just use `.filter(prod(s) > max_size)`. First-party Hypothesis strategies try to avoid scenarios where a user can define "unsatisfiable" strategies like `xps.array_shapes(min_dims=5, min_side=5, max_size=24)`.